File: ScorpioProto/Java/src/Datas/Int3.java
package Datas;
//本文件为自动生成，请不要手动修改
import java.util.*;
import Scorpio.Conversion.*;

public class Int3 implements IData {
    
    private List<Int2> _Value1;
    /**   默认值() */
    public List<Int2> getValue1() { return _Value1; }
    private Integer _Value2;
    /**   默认值() */
    public Integer getValue2() { return _Value2; }
    
    public Int3(String fileName, IReader reader) throws Exception {
        {
            List<Int2> list = new ArrayList<Int2>();
            int number = reader.ReadInt32();
            for (int i = 0; i < number; ++i) { list.add(new Int2(fileName, reader)); }
            this._Value1 = Collections.unmodifiableList(list);
        }
        this._Value2 = reader.ReadInt32();
    }
    
    public Object GetData(String key) {
        if ("Value1".equals(key)) return _Value1;
        if ("Value2".equals(key)) return _Value2;
        return null;
    }
    
    public void Set(Int3 value) {
        this._Value1 = value._Value1;
        this._Value2 = value._Value2;
    }
    
    @Override
    public String toString() {
        return "Value1:" + _Value1 + "," + "Value2:" + _Value2;
    }
}

File: ScorpioProto/CSharp/Example/src/Int3.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Conversion;

namespace Datas {
public partial class Int3 : IData {
    
    /* <summary>   默认值() </summary> */
    public ReadOnlyCollection<Int2> Value1 { get; private set; }
    /* <summary>   默认值() </summary> */
    public int Value2 { get; private set; }
    
    public Int3(string fileName, IReader reader) {
        {
            var list = new List<Int2>();
            var number = reader.ReadInt32();
            for (int i = 0; i < number; ++i) { list.Add(new Int2(fileName, reader)); }
            this.Value1 = list.AsReadOnly();
        }
        this.Value2 = reader.ReadInt32();
    }
    
    public object GetData(string key) {
        if ("Value1".Equals(key)) return Value1;
        if ("Value2".Equals(key)) return Value2;
        return null;
    }
    
    public void Set(Int3 value) {
        this.Value1 = value.Value1;
        this.Value2 = value.Value2;
    }
    
    public override string ToString() {
        return $"Value1:{Value1}, Value2:{Value2}, ";
    }
}
}
